xlators/mgmt/glusterd/src/glusterd-sm.c

```diff
@@ -1058,15 +1058,18 @@ glusterd_ac_handle_friend_add_req(glusterd_friend_sm_event_t *event, void *ctx)
 
     new_event->ctx = new_ev_ctx;
 
-    ret = dict_get_strn(ev_ctx->vols, "hostname_in_cluster",
-                        SLEN("hostname_in_cluster"), &hostname);
-    if (ret || !hostname) {
-        gf_msg_debug(this->name, 0, "Unable to fetch local hostname from peer");
-    } else if (snprintf(local_node_hostname, sizeof(local_node_hostname), "%s",
-                        hostname) >= sizeof(local_node_hostname)) {
-        gf_msg_debug(this->name, 0, "local_node_hostname truncated");
-        ret = -1;
-        goto out;
+    if (ev_ctx->vols) {
```

Comment:
There are 2 occurrences reported by coverity for the same issue. Another one is at line #1007. Don't we need to fix that as well?

Reply:
Because the other one is false positive and won't happen.
We will reach that part of code only if `status` is not `GLUSTERD_VOL_COMP_RJT`, and in that case, we are not freeing `ev_ctx->vols`.

Reply:
Yes you are right. In the code flow that is reported by Coverity, the ret value of glusterd_compare_friend_data() will be -1 and it will directly jump to out. We will not even reach line  1007 at all in that case.

Reply:
How `ev_ctx->vols` could be `NULL` ? even if it's released, no one seems to be setting it to `NULL`.

Reply:
I knew that `free()` also sets the pointer to NULL, but I just discovered that's not the case.
So, based on that if we set `peer_data = NULL` on line: 5475 in `glusterd_compare_friend_data()` in `glusterd-utils.c` I think that should work. Or we can do it in a different (better) way, what do you suggest?

Reply:
Nice catch @xhernandez, I somehow missed that it is not being set to NULL anywhere.